packages/beeq/src/components/date-picker/bq-date-picker.tsx

```diff
@@ -0,0 +1,545 @@
+  handleValueChange() {
+    if (Array.isArray(this.value)) {
+      this.hasValue = this.value.some((val) => val.length > 0);
+      return;
+    }
+
+    this.hasValue = isDefined(this.value);
+  }
+
+  @Watch('type')
+  checkPropValues() {
+    validatePropValue(DATE_PICKER_TYPE, 'single', this.el, 'type');
+  }
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  /** Callback handler emitted when the input loses focus */
+  @Event() bqBlur!: EventEmitter<HTMLBqDatePickerElement>;
+
+  /**
+   * Callback handler emitted when the input value has changed and the input loses focus.
+   * This handler is called whenever the user finishes typing or pasting text into the input field and then clicks outside of the input field.
+   */
+  @Event() bqChange!: EventEmitter<{ value: string; el: HTMLBqDatePickerElement }>;
+
+  /** Callback handler emitted when the input value has been cleared */
+  @Event() bqClear!: EventEmitter<HTMLBqDatePickerElement>;
+
+  /** Callback handler emitted when the input has received focus */
+  @Event() bqFocus!: EventEmitter<HTMLBqDatePickerElement>;
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  componentDidLoad() {
+    this.handleValueChange();
+  }
+
+  // Listeners
+  // ==============
+
+  @Listen('bqOpen', { capture: true })
+  handleOpenChange(ev: CustomEvent<{ open: boolean }>) {
+    if (!ev.composedPath().includes(this.el)) return;
+
+    this.open = ev.detail.open;
+  }
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  /**
+   * Clears the selected value.
+   *
+   * @return {Promise<void>}
+   * @memberof BqInput
+   */
+  @Method()
+  async clear(): Promise<void> {
+    if (this.disabled) return;
+
+    this.value = undefined;
+
+    this.bqClear.emit(this.el);
+  }
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private handleBlur = () => {
+    if (this.disabled) return;
+
+    this.bqBlur.emit(this.el);
+  };
+
+  private handleFocus = () => {
+    if (this.disabled) return;
+
+    this.bqFocus.emit(this.el);
+  };
+
+  private handleInput = (ev: Event) => {
+    if (this.disabled) return;
+
+    if (!isHTMLElement(ev.target, 'input')) return;
+    this.value = ev.target.value;
+  };
+
+  private handleChange = (ev: Event) => {
+    if (this.disabled) return;
+
+    if (!isHTMLElement(ev.target, 'input')) return;
+
+    const dateValue = new Date(ev.target.value);
+    if (!isNaN(dateValue.getTime())) {
+      this.value = new Intl.DateTimeFormat('fr-CA', { year: 'numeric', month: '2-digit', day: '2-digit' }).format(
```

Comment:
Why are we using `french canada` locale here ?

Reply:
Yes, we need to keep activ this format yyyy-mm-dd as well.

Reply:
I would use here [formatToParts](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/formatToParts#description) with a `filter` and `join` to construct format `yyyy-mm-dd`

Reply:
Indeed, but you need something like  the following:

```js
const dateTimeFormat = new Intl.DateTimeFormat(this.locale, {
  year: 'numeric',
  month: '2-digit',
  day: '2-digit',
});
const dateParts = dateTimeFormat.formatToParts(dateValue);
const dateObject: { year?: string; month?: string; day?: string } = dateParts.reduce((acc, part) => {
  if (part.type === 'year' || part.type === 'month' || part.type === 'day') {
    acc[part.type] = part.value;
  }
  return acc;
}, {});
this.value = `${dateObject.year}-${dateObject.month}-${dateObject.day}`;
```
wondering if it is worth tho 🤔 

We could also simplify this by just doing:

```js
if (!isNaN(dateValue.getTime())) {
  this.value = dateValue.toLocaleDateString('fr-CA');
  this.formattedDate = this.formatDate(this.value);
  this.bqChange.emit({ value: this.value, el: this.el });
}
```